docs/proposal/20240715-namespaced-iam-identities.md

```diff
@@ -0,0 +1,472 @@
+	Selector metav1.LabelSelector `json:"selector"`
+	AWSRoleSpec            `json:",inline"`
+	// A unique identifier that might be required when you assume a role in another account.
+	// If the administrator of the account to which the role belongs provided you with an
+	// external ID, then provide that value in the ExternalId parameter. This value can be
+	// any string, such as a passphrase or account number. A cross-account role is usually
+	// set up to trust everyone in an account. Therefore, the administrator of the trusting
+	// account might send an external ID to the administrator of the trusted account. That
+	// way, only someone with the ID can assume the role, rather than everyone in the
+	// account. For more information about the external ID, see How to Use an External ID
+	// When Granting Access to Your AWS Resources to a Third Party in the IAM User Guide.
+	// +optional
+	ExternalID string `json:"externalID,omitempty"`
+
+	// SourceIdentityRef is a reference to another identity which will be chained to do
+	// role assumption. All identity types are accepted.
+	SourceIdentityRef *AWSIdentityReference `json:"sourceIdentityRef,omitempty"`
+}
+```
+
+## Security considerations
+
+This proposal relies on AWS SourceIdentity field which goal is to identify the principal on behalf of which the AssumeRole action is called,
+as defined in the [aws-sdk-go-credential-provider].
+
+```golang
+// AssumeRoleOptions is the configurable options for AssumeRoleProvider
+type AssumeRoleOptions struct {
+	// [...]
+
+	// The source identity specified by the principal that is calling the AssumeRole
+	// operation. You can require users to specify a source identity when they assume a
+	// role. You do this by using the sts:SourceIdentity condition key in a role trust
+	// policy. You can use source identity information in CloudTrail logs to determine
+	// who took actions with a role. You can use the aws:SourceIdentity condition key
+	// to further control access to Amazon Web Services resources based on the value of
+	// source identity. For more information about using source identity, see Monitor
+	// and control actions taken with assumed roles
+	// (https://docs.aws.amazon.com/IAM/latest/UserGuide/id_credentials_temp_control-access_monitor.html)
+	// in the IAM User Guide.
+	SourceIdentity *string
+}
+```
+
+By the definition of this field, the `SourceIdentity` field should be managed by CAPA and not exposed to the cluster administrators in any mean.
+
+The `SourceIdentity` may be customisable by the CAPA owners to customise a certain prefix and hence increase the unicity of the requests.
+The default `SourceIdentity` field may look like `CAPA:provider:aws:AWSRoleIdentity:identity-namespace:identity-name`. The values `AWSRoleIdentity`, `identity-namespace`
+and `identity-name` refer to kubernetes resources and must be injected by the CAPA controller without any posibility to be changed by neither the CAPA owners or the cluster administrators.
```

Comment:
This isn't entirely clear to me - is the proposal that once created, the value for `SourceIdentity` is made up of values taken from the `AWSStaticIdentity` or `AWSRoleIdentity`? So this value is readable/buildable to the CAPA owners and cluster administrators because they can read the Kubernetes objects, but they would not be able to modify it?

Reply:
the value of `SourceIdentity` must be composed by the `kind` of identity, as well as their namespace and name.

for example, if you have an identity 

```yaml
kind: AWSRoleIdentity
metadata:
  namespace: my-namespace
  name: my-name
```

the `SourceIdentity` must be `CAPA:provider:aws:AWSRoleIdentity:my-namespace:my-name`. It must not be possible to change the  `AWSRoleIdentity:my-namespace:my-name` part of the `SourceIdentity` by the `AWSRoleIdentity`

Does it make it any clearer?